tests/build.rs

```diff
@@ -1790,13 +1790,15 @@ fn example_bin_same_name() {
         .unwrap();
 
     assert_that(&p.bin("foo"), is_not(existing_file()));
+    assert_that(&p.bin("foo-test"), existing_file());
```

Comment:
Hm ok so this may be where the `-test` is showing up.

This looks like a somewhat orthogonal change, however (having predictably named test artifacts). Perhaps we could leave that to a future PR?